nilearn/input_data/tests/test_nifti_labels_masker.py

```diff
@@ -368,3 +367,24 @@ def test_nifti_labels_masker_with_mask():
         masked_labels, resampling_target=None, mask_img=mask_img)
     masked_signals = masked_masker.fit().transform(fmri_img)
     assert np.allclose(signals, masked_signals)
+
+def test_3d_images():
+    # Test that the NiftiLabelsMasker works with 3D images
+    affine = np.diag((4, 4, 4, 1))
+    n_regions = 3
+    shape3 = (2, 2, 2)
+
+    labels33_img = data_gen.generate_labeled_regions(shape3, n_regions,
+                                                     affine=affine)
+    mask_img = nibabel.Nifti1Image(np.ones(shape3, dtype=np.int8),
+                           affine=affine)
+    epi_img1 = nibabel.Nifti1Image(np.ones(shape3),
+                           affine=affine)
+    epi_img2 = nibabel.Nifti1Image(np.ones(shape3),
+                           affine=affine)
+    masker = NiftiLabelsMasker(labels33_img, mask_img=mask_img)
+
+    epis = masker.fit_transform(epi_img1)
```

Comment:
You should check that the shapes of the outputs are correct. For the labels masker you should get a 2D array of shape `n_time_points x n_regions`. So, when you fit transform `epi_img1` you expect `epis` to be `1 x 3`, and when fit transforming `[epi_img1, epi_img2]` you expect `epis` to be `2 x 3`.